mlrun/utils/helpers.py

```diff
@@ -689,12 +689,14 @@ def generate_object_uri(project, name, tag=None, hash_key=None):
     return uri
 
 
-def generate_artifact_uri(project, key, tag=None, iter=None):
+def generate_artifact_uri(project, key, tag=None, iter=None, tree=None):
     artifact_uri = f"{project}/{key}"
     if iter is not None:
         artifact_uri = f"{artifact_uri}#{iter}"
     if tag is not None:
         artifact_uri = f"{artifact_uri}:{tag}"
+    if tree is not None:
+        artifact_uri = f"{artifact_uri}@{tree}"
```

Comment:
this changes the behavior where you added this.
```
        # with_tag was using tree instead of tag before the refactor,
        # so the naming is changed but the behavior is the same
```
is this a bug?